### ISSUE TITLE
Empty png and svg files

### ISSUE THREAD

**Issue by Saibamen** [sverweij/dependency-cruiser#108]
Empty png and svg images.
Windows 10.

Running `depcruise -x node_modules -T dot . | dot -T png > coffee-script-coffee-without-node_modules.png` give me 0 KB invalid file.

**Comment by sverweij** [comment]
hey @Saibamen I need a little more context to help you out with that.

Some questions that might help you in the the mean time (assuming you're running `depcruise` in the folder you want to visualize):
- What's the output of `depcruise --info` ?
- What's the output of `depcruise -x node_modules .` ? 

If your code base needs a transpiler (coffeescript I guess from the filename you're using for output?) - make sure it's installed at the same spot as dependency-cruiser is, so it can use it.

If that doesn't help don't hesitate to ping back

**Comment by hamedsbt** [comment]
what is dot? how install dot? I got error:

**Comment by sverweij** [comment]
@hamedsbt thanks for asking - I should probably add the answer to the FAQ

GraphViz dot is a program that takes care of visualising graphs. It takes a graph description written in the `dot` language as input and emits a graphical representation.

If you pass --output-type dot to dependency-cruiser, it'll emit a graph description in dot. GraphViz dot takes care of the hard algorithms to auto layout & render it.

- More information: https://www.graphviz.org
- And there is a windows version: https://www.graphviz.org/download/

You can also copy dependency-cruiser's dot output in one of the online graphviz interpreters (e.g. http://viz-js.com) or use my graphviz package for the atom editor (https://atom.io/packages/graphviz-preview-plus)

HTH

**Comment by hamedsbt** [comment]
thank you for fast answer, I deleted my question becuase I found solution. thank you again.

**Comment by Saibamen** [comment]
```Transpilers:

    √ javascript (>es1)
    × coffee-script (>=1.0.0 <2.0.0)
    × coffeescript (>=1.0.0 <3.0.0)
    × livescript (>=1.0.0 <2.0.0)
    × typescript (>=2.0.0 <4.0.0)

  Extensions:

    √ .js
    √ .mjs
    √ .jsx
    √ .vue
    × .ts
    × .tsx
    × .d.ts
    × .ls
    × .coffee
    × .litcoffee
    × .coffee.md
    × .csx
    × .cjsx```

```
√ no dependency violations found (128 modules cruised)
```

**Comment by sverweij** [comment]
@Saibamen so the regular dependency-cruise works fine. So I'm assuming it's a javascript based project (it won't scan anything else as it didn't find any of the alt-js compilers).

This means there's two possible causes, in this order:
- GraphViz dot isn't installed/ doesn't work correctly in the environment you're running it on. 
- dependency-cruiser generated a dot file that isn't valid

You can verify which one by running `depcruise -x node_modules -T dot . > dependencygraph.txt`. and pasting the outcome in an online graphviz app (e.g. http://viz-js.com). 

If that renders a picture there's something afoot with the graphviz installation.
If it doesn't I'd like to see the `dependencygraph.txt` for closer inspection 

I've attached a graphviz dot file of comparable size whose content you can drag into http://viz-js.com for reference: [dependency-cruiser-graph.txt](https://github.com/sverweij/dependency-cruiser/files/2963895/dependency-cruiser-graph.txt)

HTH

**Comment by Saibamen** [comment]
http://viz-js.com shows correctly SVG for my `.txt` file

**Comment by sverweij** [comment]
so ...
1.  dependency-cruiser emits a valid dot program 
2. the dot program renders when running it through graphviz on line
3. on windows 10, apparently piping a valid dot program into graphviz dot does _not_ work

This means there's something fishy going  on with pipes on windows 10 (I've tested on windows 7 - and that works ok). There might be limits to anonymous pipes, or windows 10 might require `\r\n` when stdout is done. I've searched for clues but little turned up ...

I don't have access to a windows 10 machine, so it's tough debugging - when I have a few hours spare time I can try my chances at a VM though...

In the mean time, as a workaround on windows 10, you can use
```msdos
depcruise -x node_modules -T dot . > dependencygraph.txt
```
and then transform the output with graphviz dot
```msdos
dot -Tsvg dependencygraph.txt -odependencygraph.svg
```

b.t.w. 
- the graphviz version you use is pretty recent, so that's probably not the problem
- I've been assuming you're using either cmd.exe or the powershell - let me know if it's something different

**Comment by sverweij** [comment]
I've been able to reproduce the issue on windows 7 - using version 2.41 as available on the graphviz website - version 2.41 is the _development_ release - and it seems to lack some basic features (like the dot layout engine and support for _any_ output format). Running `dot` immediately returns an error - 

After that I installed the _stable_ version (2.38) and that worked without a hitch.

HTH
